src/audio.c

```diff
@@ -629,6 +631,27 @@ static void check_telev(struct audio *a, struct autx *tx)
 }
 
 
+static bool silence(const int16_t *sampv, size_t sampc)
+{
+#ifdef USE_SILENCE_DETECTION
+	int32_t sum = 0;
+	size_t i;
+
+	for (i = 0; i < sampc; i++) {
+		sum += sampv[i]*sampv[i];
+
+		if (sum > (int32_t) (i+1) * SILENCE_INT16)
+			return false;
+	}
+#else
+	(void) sampv;
+	(void) sampc;
+#endif
+
+	return true;
+}
+
+
```

Comment:
would it be possible to use this API for silence detection:

`double aulevel_calc_dbov(int fmt, const void *sampv, size_t sampc);`

it supports more audio sample formats, and it also calculates the
correct energy/power level of the signal.